### PR TITLE
SSO Booster Deployment and Interaction

### LINKED ISSUE #717: Secured mission: ZIP booster deployment workflow
* Remove Launcher workflow
* add ZIP deployment workflow
** can be adapted from the existing ZIP workflow, if the SSO application is deployed from a git submodure
* Steps to manually deploy both applications to OSO Pro and OSCP
** Include note that the booster can not be deployed available on OSO Starter

### PR DESCRIPTION
**Pull request by tradej**
This is the deployment and interaction rework of the SSO mission in all runtimes.

Notes:
* I had to tweak the modules used also in the REST API deployment to make them reusable in the SSO section.
* The `:runtime-tag:` attribute was removed from the runtime guide master files in favour of the `:parameter-runtime:` attribute, to be consistent with parametre usage across modules.
* I have adapted the information from the [Google Doc](https://docs.google.com/document/d/1mum_guluetn9faHwyopU1QDWyNEfMG4gnAolR_cP40o/edit#heading=h.ikoz6p3uetbz) for all runtimes (it is originally written for Vert.x). Please check if the commands etc. are factually correct.
* I have deleted the information about deploying the SSO booster using the launcher. If we want to reintroduce it in the future, we should repurpose the REST API deployment module (`docs/topics/proc_deploying-the-booster-using-launcher.adoc`).
* Resolves #717.

Please also review the following, rendered, chapters:

* _Building and deploying the Secured booster to Single-node OpenShift Cluster_
* _Building and Deploying the Secured Booster to the OpenShift Container Platform_
* _Interacting with the Secured Booster_

of the following guides:

* [Eclipse Vert.x Runtime Guide](http://download.eng.brq.redhat.com/scratch/tradej/rhoar/issue-717/vertx-runtime.html#building-and-deploying-the-secured-booster-to-openshiftlocal)
* [Node.js Runtime Guide](http://download.eng.brq.redhat.com/scratch/tradej/rhoar/issue-717/nodejs-runtime.html#building-and-deploying-the-secured-booster-to-openshiftlocal)
* [Spring Boot Tomcat Runtime Guide](http://download.eng.brq.redhat.com/scratch/tradej/rhoar/issue-717/spring-boot-runtime.html#building-and-deploying-the-secured-booster-to-openshiftlocal)
* [WildFly Swarm Runtime Guide](http://download.eng.brq.redhat.com/scratch/tradej/rhoar/issue-717/wf-swarm-runtime.html#building-and-deploying-the-secured-booster-to-openshiftlocal)

The links point to the first chapter to be reviewed in the respective book, and the chapters follow each other, so you can just keep reading until you encounter the _Secured SSO Resources_ chapter.

### CODE REVIEW

File: docs/topics/assembly_interacting-with-the-secured-booster.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// This assembly uses a parameterized module. Parameters used:
+//
+//   parameter-runtime: A segment of the endpoint name
+//
+// Rationale: The API endpoint procedure is identical on all runtimes
+
+[#interacting-with-the-secured-booster]
+= Interacting with the {name-mission-secured} Booster
+
+The Secured booster provides a default HTTP endpoint that accepts `GET` requests if the caller is authenticated and authorized.
```
Comment (Ladicek) [inline docs/topics/assembly_interacting-with-the-secured-booster.adoc]: Should the beginning of this sentence also use `{name-mission-secured}`? (I don't know myself.)

File: docs/topics/assembly_interacting-with-the-secured-booster.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+The client first authenticates against the {RHSSO} server and then performs a `GET` request against the {name-mission-secured} booster using the access token returned by the authentication step.
+
+The `sso` directory in the ZIP archive with the booster contains a Java client you can use to interact with the {name-mission-secured} booster.
```
Comment (Ladicek) [inline docs/topics/assembly_interacting-with-the-secured-booster.adoc]: There's no `sso` directory anymore.

File: docs/topics/proc_deploying-the-secured-booster-using-the-oc-cli-client.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+$ oc get route secure-sso -o jsonpath='{"https://"}{.spec.host}{"/auth\n"}'
+----
+
+ifndef::http-api-nodejs[]
+. Use Maven to start the deployment to {OpenShiftLocal}.
++
+--
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ mvn clean fabric8:deploy -Popenshift -DskipTests -DSSO_AUTH_SERVER_URL=<SSO_AUTH_SERVER_URL>
```
Comment (Ladicek) [inline docs/topics/proc_deploying-the-secured-booster-using-the-oc-cli-client.adoc]: I know this works with Swarm, but someone from Vert.x and Spring Boot should review as well.

**Comment by Ladicek** [comment]
I don't know where it comes from, but I see stuff like

> To work with the Fabric8 Launch application on OpenShift Container Platform using the oc command-line client, you need to authenticate it using the token the Fabric8 Launch application provides you.

which doesn't make any sense. And the following content doesn't really say anything about Fabric8 Launch. This is purely about logging into OCP on the cmdline, using `oc`.

File: docs/topics/proc_interacting-with-the-secured-booster-using-the-web-interface.adoc [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+.The Result of an Authenticated Greeting Request (as Alice)
+image::sso-invoke-alice.png[]
+--
+.. Log out.
+
+. Perform an authenticated request as an admininstrator:
+.. Click the _Invoke_ button.
++
+Confirm that this sends an unauthenticated request to the Greeting service.
+.. Click th _Login_ button and log in as xref:sso-users[the _admin_ user].
```
Comment (Ladicek) [inline docs/topics/proc_interacting-with-the-secured-booster-using-the-web-interface.adoc]: "Click th"

**Comment by Ladicek** [comment]
There also seems to be a certain overlap between this and #743; could you guys sync up on this?

**Comment by Ladicek** [comment]
Otherwise LGTM.

File: docs/topics/assembly_interacting-with-the-secured-booster.adoc (hunk 1 above)
Comment (tradej) [reply docs/topics/assembly_interacting-with-the-secured-booster.adoc]: Thanks, good catch, it should.

File: docs/topics/assembly_interacting-with-the-secured-booster.adoc (hunk 2 above)
Comment (tradej) [reply docs/topics/assembly_interacting-with-the-secured-booster.adoc]: Thanks, will correct.

File: docs/topics/proc_interacting-with-the-secured-booster-using-the-web-interface.adoc (hunk 4 above)
Comment (tradej) [reply docs/topics/proc_interacting-with-the-secured-booster-using-the-web-interface.adoc]: Thanks, will correct.

**Comment by tradej** [comment]
> which doesn't make any sense. And the following content doesn't really say anything about Fabric8 Launch. This is purely about logging into OCP on the cmdline, using oc.

I've changed it to:

* _To work with boosters on Single-node OpenShift Cluster using the oc command-line client, you need to authenticate the client using the token provided by the Fabric8 Launch application._ for OSL.
* _To work with boosters on OpenShift Container Platform using the oc command-line client, you need to authenticate the client using the token provided by the OpenShift Container Platform web interface._ for OCP.

Is that better, @Ladicek?

**Comment by tradej** [comment]
@Ladicek, can it say _provided by the Single-node OpenShift Cluster web console,_ and _provided by the OpenShift Container Platform web console,_ respectively?

**Comment by Ladicek** [comment]
That works for me.

**Comment by tradej** [comment]
@Ladicek, thanks.

**Comment by tradej** [comment]
Changes done, only the Vert.x and Spring Boot deployment command review is pending. I have requested it on Slack.

**Comment by tradej** [comment]
One review is in, the command is correct for Vert.x. Spring Boot still pending.

**Comment by rhoads-zach** [comment]
merged: 5f2827ede3fd5dac3cc515749e847f096a871ecd

**Pull request closed without merging**